watertap/unit_models/nanofiltration_ZO.py

```diff
@@ -46,6 +47,12 @@
 _log = idaeslog.getLogger(__name__)
 
 
+@deprecated(
+    "The NanofiltrationZO model has been deprecated in favor of the Nanofiltration0D model. "
+    "The Nanofiltation0D model has most of the capabilities of the NanofiltrationZO model "
```

Comment:
Thanks for this! You say the new model has "most of the capabilities". Can you remind me what the new model might _not_ have that the deprecated version _does_? It's been a while since I reviewed this PR.

Reply:
Off the top of my head, it lacks the water/solvent flux and recovery calculations. That might be all however.

Reply:
If we are to replace the deprecated version with this one eventually (and remove the deprecated later), we should probably preserve solvent flux, recovery, and anything else that might be missing.

Reply:
The first catch I see there is that to have solvent flux requires additional user input that is not required for the simplest version (at a minimum, membrane area). So, it does open the question of whether we want to have that as a default inclusion, or if it should be an optional add-on.

Also, I am not sure how much time I will have to work on this right now. Someone else is welcome to take over if we decide it is important to add the other components.

Reply:
I think that we should include flux, area, and other basic "essentials" that are in the existing NF model. 

We can have someone else ensure those additions are made and then merge this in.

Reply:
@adam-a-a do we also want to remove/deprecate the _other_ ZO NF model (i.e.,  `unit_models/zero_order/nanofiltration_zo.py`)? Once this is merged in, we would have four NF models, and two with the same name.

Reply:
@kurbansitterley because of other model dependencies, we planned to keep the other ZO formulation that you referred to and deprecate the duplicate one that the current PR's NF model is meant to replace here. Ultimately, we'd end up with one ZO model, this simple 0d formulation, and then the multicomponent DSPMDE model--also 0d but more complex than this formulation.

Does that make sense?

Reply:
Yes, that makes sense. I assumed the other ZO model is used in some flowsheets somewhere but didn't check.